{
  try {
    final Future future=getContext().getInvocationService().invokeOnRandomTarget(req);
    return context.getSerializationService().toObject(future.get());
  }
 catch (  Exception e) {
    throw ExceptionUtil.rethrow(e);
  }
}
